src/analyses/cfg_dominators.h

```diff
@@ -16,20 +16,163 @@ Author: Georg Weissenbacher, georg@weissenbacher.name
 #include <list>
 #include <map>
 #include <iosfwd>
-#include <cassert>
+#include <algorithm>
+#include <fstream>
+#include <stack>
+#include <tuple>
+#include <type_traits>
+#include <iterator>
 
 #include <goto-programs/goto_functions.h>
 #include <goto-programs/goto_program.h>
 #include <goto-programs/cfg.h>
 
-template <class P, class T, bool post_dom>
-class cfg_dominators_templatet
-{
-public:
-  typedef std::set<T> target_sett;
+template<typename T, typename node_indext>
+struct dominators_datat {
+  std::vector<T> data;
+  std::vector<node_indext> immediate_dominator;
+};
+
+/// An immutable set of dominators. Constant memory usage and creation time, but linear lookup time
+/// Immutability is necessary because the structure uses sharing
+template<typename T, typename node_indext>
+struct dominatorst {
+  using datat = dominators_datat<T, node_indext>;
+  static constexpr node_indext npos = std::numeric_limits<node_indext>::max();
```

Comment:
As far as I recall that doesn't work with Visual Studio.

Reply:
It does work with VS2015+; In this case this can just be changed to `const` but there was some discussion on moving the minimum requirement to VS2015.

Reply:
If that can be done for AppVeyor then surely that's fine - but for now the build fails at that line. Or is it just a missing `#include <limits>`?